### PR TITLE
Type memo summary

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
Add typing and a type guard specifically for transaction summaries containing a transfer with a memo.

## Changes
- Ran `lint --fix`
- Updated types for transfer with memo and added a type guard.
- Added an example of how to extract the memo and toAddress from a block containing a transfer with a memo.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -207,6 +207,26 @@ const bestBlock = consensusStatus.bestBlock;
...
 
+## Check block for transfers with memo
+The following example demonstrates how to check and parse a block 
+for transfers with a memo.
+```
+const blockHash = "b49bb1c06c697b7d6539c987082c5a0dc6d86d91208874517ab17da752472edf";
+const blockSummary = await client.getBlockSummary(blockHash);
+const transactionSummaries = blockSummary.transactionSummaries;
+
+for (let transactionSummary of transactionSummaries) {
```
Comment (shjortConcordium) [inline README.md:218]:
```suggestion
for (const transactionSummary of transactionSummaries) {
```

File: README.md [hunk 2]
```diff
@@ -207,6 +207,26 @@ const bestBlock = consensusStatus.bestBlock;
...
+    if (instanceOfTransferWithMemoTransactionSummary(transactionSummary)) {
+        const [transferredEvent, memoEvent] = transactionSummary.result.events;
+
+        const toAddress = transferredEvent.to.address;
+        const memo = memoEvent.memo;
+
+        // Apply business logic to toAddress and memo...
```
Comment (shjortConcordium) [inline README.md:225]: Maybe show the amount can be extracted too?

File: src/types.ts [hunk 3]
```diff
@@ -94,34 +93,73 @@ export interface TransferredEvent {
...
+    | GenericTransactionSummary
+    | TransferWithMemoTransactionSummary;
+
+export function instanceOfTransferWithMemoTransactionSummary(
+    object: TransactionSummary
+): object is TransferWithMemoTransactionSummary {
+    return (
+        object.type !== undefined &&
+        object.type.contents !== undefined &&
+        object.type.contents === 'transferWithMemo'
```
Comment (shjortConcordium) [inline src/types.ts:159]: `object.type.contents !== undefined` is redundant.

**Review by shjortConcordium — COMMENTED** [review]
Please add a test for similar to the example.

**Comment by orhoj** [comment]
> Please add a test for similar to the example.

I have added a test case that uses the new functionality.

**Review by shjortConcordium — APPROVED** [review]

**Pull request merged**